Modules/_threadmodule.c

```diff
@@ -2407,6 +2411,30 @@ PyDoc_STRVAR(shutdown_doc,
 \n\
 Wait for all non-daemon threads (other than the calling thread) to stop.");
 
+/* gh-123940: Mark remaining daemon threads as exited so that they may
+ * be joined from finalizers.
+ */
+void
+_PyThread_DaemonThreadsForceKilled(PyInterpreterState *interp)
```

Comment:
Likewise, let's move this function to Python/thread.c.

Reply:
Eric and I chatted in person. This would involve a larger change that moves ThreadHandle and associated functions. We want this to be a targeted fix, so that should happen in a follow on PR.